src/common.rs

```diff
@@ -53,12 +53,14 @@ fn metadata_command(args: MetadataArgs) -> MetadataCommand {
     command
 }
 
-pub fn sourced_dependencies(metadata_args: MetadataArgs) -> Vec<SourcedPackage> {
+pub fn sourced_dependencies(
+    metadata_args: MetadataArgs,
+) -> Result<Vec<SourcedPackage>, anyhow::Error> {
     let command = metadata_command(metadata_args);
     let meta = match command.exec() {
         Ok(v) => v,
-        Err(cargo_metadata::Error::CargoMetadata { stderr: e }) => err_exit(&e),
-        Err(err) => err_exit(format!("Failed to fetch crate metadata!\n  {}", err).as_str()),
+        Err(cargo_metadata::Error::CargoMetadata { stderr: e }) => return Err(anyhow!(e)),
```

Comment:
`anyhow` provides a convenience macro for this:

```suggestion
        Err(cargo_metadata::Error::CargoMetadata { stderr: e }) => bail!(e),
```

Please apply this to other occurrences of `return Err(anyhow!(...))`